src/types/index.ts

```diff
@@ -1,46 +1,408 @@
-// @ts-ignore
-interface AutoComplete {}
+// -----------------------------------------------------------------------------
+//  --------------------------------- TYPES ------------------------------------
+// -----------------------------------------------------------------------------
 
-interface StopGroup {}
+type GroundPlacesObject = Record<Gpuid, StopGroup | StopCluster>;
 
-interface StopCluster {}
+type Gpuid = string;
 
-interface StopGroupProperties {}
+type StopGroupGpuid = Gpuid;
 
-interface StopClusterProperties {}
+type StopClusterGpuid = Gpuid;
 
-interface Filters {}
+// -----------------------------------------------------------------------------
+//  ------------------------------- INTERFACES ---------------------------------
+// -----------------------------------------------------------------------------
 
-interface GroundPlacesDiff {}
+interface StopGroup {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.GROUP;
+  childs: SegmentProviderStop[];
+  serviced: Serviced;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type Gpuid = string;
+interface StopCluster {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.CLUSTER;
+  childs: StopGroupGpuid[];
+  serviced: Serviced;
+  unique_name?: string;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type StopGroupGpuid = Gpuid;
+interface StopGroupInfos {
+  countryCode: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType;
+  serviced: Serviced;
+}
 
-type StopClusterGpuid = Gpuid;
+interface StopClusterInfos {
+  countryCode: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType;
+  createdFromStopGroupGpuid: StopGroupGpuid;
+  serviced: Serviced;
+}
+
+interface StopGroupProperties {
+  name?: string;
+  longitude?: number;
+  latitude?: number;
+}
+
+interface StopClusterProperties {
```

Comment:
Messy uses of types : 
if they have the same properties why not call them GroundPlaceProps ?
adn why do we need a specific type for them ?

Reply:
Today they have the same properties, but maybe in the future the properties will change, and then the distinction will make sense, I implemented it this way but if you think it's better to group them together why not

Reply:
As i just said : futur is futur. Let's focus on present ;)

Reply:
After call, this is now grouped